src/TesApi.Web/AzureProxy.cs

```diff
@@ -57,34 +57,38 @@ public class AzureProxy : IAzureProxy
         private readonly string azureOfferDurableId;
         private readonly string batchResourceGroupName;
         private readonly string batchAccountName;
-
+        private AzureEnvironment azEnv;
 
         /// <summary>
         /// The constructor
         /// </summary>
+        /// <param name="env">Azure Environment</param>
         /// <param name="batchAccountName">Batch account name</param>
         /// <param name="azureOfferDurableId">Azure offer id</param>
         /// <param name="logger">The logger</param>
-        public AzureProxy(string batchAccountName, string azureOfferDurableId, ILogger logger)
+        public AzureProxy(AzureEnvironment env, string batchAccountName, string azureOfferDurableId, ILogger logger)
         {
+            this.azEnv = env;
             this.logger = logger;
             this.batchAccountName = batchAccountName;
-            var (SubscriptionId, ResourceGroupName, Location, BatchAccountEndpoint) = FindBatchAccountAsync(batchAccountName).Result;
+            var (SubscriptionId, ResourceGroupName, Location, BatchAccountEndpoint) = FindBatchAccountAsync(env, batchAccountName).Result;
             batchResourceGroupName = ResourceGroupName;
             subscriptionId = SubscriptionId;
             location = Location;
-            batchClient = BatchClient.Open(new BatchTokenCredentials($"https://{BatchAccountEndpoint}", () => GetAzureAccessTokenAsync("https://batch.core.windows.net/")));
+            batchClient = BatchClient.Open(new BatchTokenCredentials($"https://{BatchAccountEndpoint}", () => GetAzureManagementAccessTokenAsync(this.azEnv)));
 
             getBatchAccountFunc = async () =>
-                await new BatchManagementClient(new TokenCredentials(await GetAzureAccessTokenAsync())) { SubscriptionId = SubscriptionId }
+                await new BatchManagementClient(new Uri(this.azEnv.ResourceManagerEndpoint),
+                new TokenCredentials(await GetAzureAccessTokenAsync(env))) { SubscriptionId = SubscriptionId }
                     .BatchAccount
                     .GetAsync(ResourceGroupName, batchAccountName);
 
             this.azureOfferDurableId = azureOfferDurableId;
 
             if (!AzureRegionUtils.TryGetBillingRegionName(location, out billingRegionName))
             {
-                logger.LogWarning($"Azure ARM location '{location}' does not have a corresponding Azure Billing Region.  Prices from the fallback billing region '{DefaultAzureBillingRegionName}' will be used instead.");
+                logger.LogWarning($@"Azure ARM location '{location}' does not have a corresponding Azure Billing Region.
```

Comment:
what was the change here?  let's not make any changes that aren't directly related to supporting the sovereign cloud, otherwise it adds noise to the PR

Reply:
It's noise, removed in new commit.